From 367fc710a94d802ace273c7aebb8f5f96647370b Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Mon, 24 Jun 2019 14:45:50 +0800
Subject: [PATCH] tweak iso mount dir and kernel name

- Tweak iso mount dir based on WRLinux

- Tweak kernel name based on OE

- Tweak nfs mount point of base repository to "/media/source"

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>

* update context for anaconda 32
* backport patch from Upstream to 'Handle bytes and strings from RPM'
* replace obsolete function 'util.getSysroot'

Signed-off-by: Kai Kang <kai.kang@windriver.com>

Rebase for anaconda 34

Signed-off-by: Kai Kang <kai.kang@windriver.com>

Rebase for anaconda 37

Signed-off-by: Kai Kang <kai.kang@windriver.com>

Rebase for anaconda 39.

* variable INSTALL_TREE has been removed in upstream

Signed-off-by: Kai Kang <kai.kang@windriver.com>

---
 pyanaconda/core/constants.py                     |  2 +-
 pyanaconda/modules/payloads/payload/dnf/utils.py | 15 +++++++++++----
 2 files changed, 12 insertions(+), 5 deletions(-)

diff --git a/pyanaconda/core/constants.py b/pyanaconda/core/constants.py
index b8075da7e3..350e7f7d94 100644
--- a/pyanaconda/core/constants.py
+++ b/pyanaconda/core/constants.py
@@ -45,7 +45,7 @@ ANACONDA_CLEANUP = "anaconda-cleanup"
 LIVE_MOUNT_POINT = "/run/initramfs/live"
 
 # Source mount points.
-MOUNT_DIR = "/run/install"
+MOUNT_DIR = "/media/realroot"
 DRACUT_ISODIR = "/run/install/source"
 DRACUT_REPO_DIR = "/run/install/repo"
 ISO_DIR = MOUNT_DIR + "/isodir"
diff --git a/pyanaconda/modules/payloads/payload/dnf/utils.py b/pyanaconda/modules/payloads/payload/dnf/utils.py
index 196632cbfe..9e0c8805a9 100644
--- a/pyanaconda/modules/payloads/payload/dnf/utils.py
+++ b/pyanaconda/modules/payloads/payload/dnf/utils.py
@@ -62,7 +62,7 @@ def get_kernel_package(dnf_manager, exclude_list):
         return None
 
     # Get the kernel packages.
-    kernels = ["kernel"]
+    kernels = ["kernel-image"]
 
     # ARM systems use either the standard Multiplatform or LPAE platform.
     if is_lpae_available():
@@ -166,19 +166,26 @@ def get_kernel_version_list():
 
     # Find all installed RPMs that provide 'kernel'.
     ts = rpm.TransactionSet(conf.target.system_root)
-    mi = ts.dbMatch('providename', 'kernel')
+    mi = ts.dbMatch()
+    mi.pattern('name', rpm.RPMMIRE_GLOB, 'kernel-image*')
 
     for hdr in mi:
+        log.info("hdr.filenames %s" % hdr.filenames)
+
         # Find all /boot/vmlinuz- files and strip off vmlinuz-.
         files.extend((
             f.split("/")[-1][8:] for f in hdr.filenames
-            if fnmatch.fnmatch(f, "/boot/vmlinuz-*") or
-            fnmatch.fnmatch(f, "/boot/efi/EFI/%s/vmlinuz-*" % efi_dir)
+            if ((fnmatch.fnmatch(f, "/boot/*") or
+                 fnmatch.fnmatch(f, "/boot/efi/EFI/%s/*" % efi_dir))
+                and len(f.split("/")[-1]) > 7
+                and not f.endswith('.sig'))
         ))
 
     # Sort the kernel versions.
     sort_kernel_version_list(files)
 
+    log.info("kernelVersionList %s" % files)
+
     return files
 
 
-- 
2.7.4
